OutOfSchool/OutOfSchool.WebApi/Services/Implementation/OrganizationService.cs

```diff
@@ -0,0 +1,134 @@
+        {
+            if (organization == null)
+            {
+                throw new ArgumentNullException(nameof(organization), "Organization was null.");
+            }
+
+            Organization newOrganization = this.mapper.Map<OrganizationDTO, Organization>(organization);
+          
+
+            if(OrganizationRepository.IsUnique(newOrganization))
+            {
+               
+                throw new ArgumentNullException(nameof(newOrganization), "There is already an organization with such data");
+            }
+            else
+            {
+                var organization_ = await this.OrganizationRepository.Create(newOrganization).ConfigureAwait(false);
+                return await Task.FromResult(this.mapper.Map<Organization, OrganizationDTO>(organization_)).ConfigureAwait(false);
+            }         
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<OrganizationDTO> GetAll()
+        {
+            IEnumerable<OrganizationDTO> organizationDTO = this.OrganizationRepository.GetAll().Select(
+                x => this.mapper.Map<Organization, OrganizationDTO>(x));
+
+            return organizationDTO;
+        }
+
+        /// <inheritdoc/>
+        public async Task<OrganizationDTO> GetById(long id)
+        {
+            Organization organization = this.OrganizationRepository.GetById(id).Result;
+            if (organization == null)
+            {
+                throw new ArgumentException("Incorrect Id!", nameof(id));
+            }
+
+            return await Task.Run(() =>
+            {
+                return this.mapper.Map<Organization, OrganizationDTO>(organization);
+            }).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc/>
+        public void Update(OrganizationDTO organizationDTO)
+        {
+            if (organizationDTO == null)
+            {
+                throw new ArgumentNullException(nameof(organizationDTO), "Organization was null.");
+            }
+
+            if (organizationDTO.EDRPOU.Length == 0)
+            {
+                throw new ArgumentException("EDRPOU code is empty", nameof(organizationDTO));
+            }
+
+            if (organizationDTO.INPP.Length == 0)
+            {
+                throw new ArgumentException("INPP code is empty", nameof(organizationDTO));
+            }
+
+            if (organizationDTO.MFO.Length == 0)
+            {
+                throw new ArgumentException("MFO code is empty", nameof(organizationDTO));
+            }
+
+            if (organizationDTO.Title.Length == 0)
+            {
+                throw new ArgumentException("Title is empty", nameof(organizationDTO));
+            }
+
+            if (organizationDTO.Description.Length == 0)
+            {
+                throw new ArgumentException("Description is empty", nameof(organizationDTO));
+            }
+
+            Organization organization = this.mapper.Map<OrganizationDTO, Organization>(organizationDTO);
```

Comment:
consider adding extension methods for mapping, then you can write this conversion as organizationDTO.ToDomain()